src/Cache.zig

```diff
@@ -12,6 +12,15 @@ const mem = std.mem;
 const fmt = std.fmt;
 const Allocator = std.mem.Allocator;
 
+/// Process-scoped map keeping track of all locked Cache hashes, to detect deadlocks.
+/// The plan is to enable this for debug builds only, but for now we enable
+/// it always to catch a deadlock.
```

Comment:
This is now enabled only for debug builds as of 2b97b75bb073a691400297a56daa6f42480020c7